hpedockerplugin/volume_manager.py

```diff
@@ -434,19 +515,332 @@ def get_path(self, volname):
+        path = FilePath(device_info['path']).realpath()
+        if path.exists is False:
+            msg = (_('path: %s,  does not exist'), path)
+            LOG.error(msg)
+            raise exception.HPEPluginMountException(reason=msg)
+
+        LOG.debug('path for volume: %(name)s, was successfully created: '
+                  '%(device)s realpath is: %(realpath)s',
+                  {'name': volname, 'device': device_info['path'],
+                   'realpath': path.path})
+
+        # Create filesystem on the new device
+        if fileutil.has_filesystem(path.path) is False:
+            fileutil.create_filesystem(path.path)
+            LOG.debug('filesystem successfully created on : %(path)s',
+                      {'path': path.path})
+
+        # Determine if we need to mount the volume
+        if vol_mount == volume.DEFAULT_MOUNT_VOLUME:
+            # mkdir for mounting the filesystem
+            mount_dir = fileutil.mkdir_for_mounting(device_info['path'])
+            LOG.debug('Directory: %(mount_dir)s, '
+                      'successfully created to mount: '
+                      '%(mount)s',
+                      {'mount_dir': mount_dir, 'mount': device_info['path']})
+
+            # mount the directory
+            fileutil.mount_dir(path.path, mount_dir)
+            LOG.debug('Device: %(path) successfully mounted on %(mount)s',
+                      {'path': path.path, 'mount': mount_dir})
+
+            # TODO: find out how to invoke mkfs so that it creates the
+            # filesystem without the lost+found directory
+            # KLUDGE!!!!!
+            lostfound = mount_dir + '/lost+found'
+            lfdir = FilePath(lostfound)
+            if lfdir.exists and fileutil.remove_dir(lostfound):
+                LOG.debug('Successfully removed : '
+                          '%(lost)s from mount: %(mount)s',
+                          {'lost': lostfound, 'mount': mount_dir})
+        else:
+            mount_dir = ''
+
+        path_info = {}
+        path_info['name'] = volname
+        path_info['path'] = path.path
+        path_info['device_info'] = device_info
+        path_info['connection_info'] = connection_info
+        path_info['mount_dir'] = mount_dir
+
+        self._etcd.update_vol(volid, 'path_info', json.dumps(path_info))
+
+        response = json.dumps({u"Err": '', u"Name": volname,
+                               u"Mountpoint": mount_dir,
+                               u"Devicename": path.path})
+        return response
+
+    def unmount_volume(self, volname, vol_mount, mount_id):
+        vol = self._etcd.get_vol_byname(volname)
+        if vol is None:
+            msg = (_LE('Volume unmount name not found %s'), volname)
+            LOG.error(msg)
+            raise exception.HPEPluginUMountException(reason=msg)
+
+        volid = vol['id']
+
+        ######
+        # Start of volume fencing changes
+        LOG.debug('WILLIAM %s ' % (vol))
+        if 'node_mount_info' in vol:
```

Comment:
i think node_mount_info type here is unicode and node_mount_info[self.node_id] might fail, can you please clarify